### PR TITLE
:bug: Source Gitlab: fix undeclared fields and validation errors

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
Certification:
- validation error: 
    - `epic_issues - due_date: invalid_date-time, start_date: invalid_date-time`
    - `project_members - expires_at: invalid_date-time`
- undeclared fields in deployments, epics, epic issues, groups, groups members, issues, projects, projects members, users 

## How
- updated format as "date"
- added undeclared fields.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-gitlab test report (commit [`7217b7437f`](https://github.com/airbytehq/airbyte/pull/31339/commits/7217b7437f4603414fb2370d87577e85746f239c)) - ❌

⏲️  Total pipeline duration: 09mn32s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Build source-gitlab docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ❌        |
| Code format checks                                            | ✅        |
| Validate metadata for source-gitlab                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-gitlab_fix-undeclared-fields-and-validation-errors/1697111507/7217b7437f4603414fb2370d87577e85746f239c/source-gitlab/1.8.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7217b7437f4603414fb2370d87577e85746f239c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-gitlab test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-gitlab test report (commit [`bbc6437372`](https://github.com/airbytehq/airbyte/pull/31339/commits/bbc6437372f3ababfdfe809ea06d9f275ab72342)) - ❌

⏲️  Total pipeline duration: 09mn54s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Build source-gitlab docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ❌        |
| Code format checks                                            | ✅        |
| Validate metadata for source-gitlab                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-gitlab_fix-undeclared-fields-and-validation-errors/1697112438/bbc6437372f3ababfdfe809ea06d9f275ab72342/source-gitlab/1.8.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:bbc6437372f3ababfdfe809ea06d9f275ab72342)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-gitlab test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-gitlab test report (commit [`5ed1259b34`](https://github.com/airbytehq/airbyte/pull/31339/commits/5ed1259b34eedc3d62f6508c854b446e22136ebd)) - ✅

⏲️  Total pipeline duration: 13mn20s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Build source-gitlab docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ✅        |
| Code format checks                                            | ✅        |
| Validate metadata for source-gitlab                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-gitlab_fix-undeclared-fields-and-validation-errors/1697197176/5ed1259b34eedc3d62f6508c854b446e22136ebd/source-gitlab/1.8.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5ed1259b34eedc3d62f6508c854b446e22136ebd)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-gitlab test
```

File: docs/integrations/sources/gitlab.md [hunk 1]
```diff
@@ -111,7 +111,8 @@ Gitlab has the [rate limits](https://docs.gitlab.com/ee/user/gitlab_com/index.ht
 ## Changelog
 
 | Version | Date       | Pull Request                                             | Subject                                                                                    |
-| :------ | :--------- | :------------------------------------------------------- | :----------------------------------------------------------------------------------------- |
+|:--------|:-----------|:---------------------------------------------------------|:-------------------------------------------------------------------------------------------|
+| 1.8.0   | 2023-10-12 |                                                          | Add undeclared fields to streams schemas, validate date/date-time format in stream schemas |
```
Comment (bazarnov) [inline docs/integrations/sources/gitlab.md:115]: Please, fix the changelog urls/pr

**Review by bazarnov — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-gitlab test report (commit [`4c7ff0728a`](https://github.com/airbytehq/airbyte/pull/31339/commits/4c7ff0728a37aa5ff9bf74c64940b293eb0647ad)) - ❌

⏲️  Total pipeline duration: 09mn42s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Build source-gitlab docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ❌        |
| Code format checks                                            | ✅        |
| Validate metadata for source-gitlab                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-gitlab_fix-undeclared-fields-and-validation-errors/1697206026/4c7ff0728a37aa5ff9bf74c64940b293eb0647ad/source-gitlab/1.8.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:4c7ff0728a37aa5ff9bf74c64940b293eb0647ad)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-gitlab test
```

**Comment by bazarnov** [comment]
`Refresh Token` is expired, please update creds.

**Comment by airbyte-oss-build-runner** [comment]
##  source-gitlab test report (commit [`4c7ff0728a`](https://github.com/airbytehq/airbyte/pull/31339/commits/4c7ff0728a37aa5ff9bf74c64940b293eb0647ad)) - ✅

⏲️  Total pipeline duration: 08mn06s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Build source-gitlab docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ✅        |
| Code format checks                                            | ✅        |
| Validate metadata for source-gitlab                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/daryna_source-gitlab_fix-undeclared-fields-and-validation-errors/1697207134/4c7ff0728a37aa5ff9bf74c64940b293eb0647ad/source-gitlab/1.8.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:4c7ff0728a37aa5ff9bf74c64940b293eb0647ad)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-gitlab test
```

**Pull request merged**